cephbackupstorage/cephbackupstorage/cephagent.py

```
@@ -1072,6 +1115,128 @@ def cancel(self, req):
+    def direct_upload_file(self, req):
+        try:
+            UploadHandler(req, self.upload_tasks).handle_upload()
+        except Exception as e:
+            logger.exception("File upload failed: %s", str(e))
+
+    @replyerror
+    def get_upload_file_progress(self, req):
+        cmd = jsonobject.loads(req[http.REQUEST_BODY])
+        task = self.upload_tasks.get_task(cmd.taskUuid)
+        if task is None:
+            raise Exception('file not found')
```

Ruff (0.14.10)

Create your own exception

(TRY002)